src/streamlink/plugins/artetv.py

```diff
@@ -40,34 +19,49 @@
-        variant, variantname = min([(stream["versionProg"], stream["versionLibelle"]) for stream in streams.values()],
-                                   key=itemgetter(0))
-        log.debug(f"Using the '{variantname}' stream variant")
-        for sname, stream in streams.items():
-            if stream["versionProg"] == variant:
-                if stream["mediaType"] == "hls":
-                    try:
-                        streams = HLSStream.parse_variant_playlist(self.session, stream["url"])
-                        yield from streams.items()
-                    except OSError as err:
-                        log.warning(f"Failed to extract HLS streams for {sname}/{stream['versionLibelle']}: {err}")
+    API_URL = "https://api.arte.tv/api/player/v2/config/{0}/{1}"
+    API_TOKEN = "MzYyZDYyYmM1Y2Q3ZWRlZWFjMmIyZjZjNTRiMGY4MzY4NzBhOWQ5YjE4MGQ1NGFiODJmOTFlZDQwN2FkOTZjMQ"
 
     def _get_streams(self):
-        language = self.match.group('language')
-        video_id = self.match.group('video_id')
-        if video_id is None:
-            json_url = JSON_LIVE_URL.format(language)
-        else:
-            json_url = JSON_VOD_URL.format(language, video_id)
-        res = self.session.http.get(json_url)
-        video = self.session.http.json(res, schema=_video_schema)
+        language = self.match.group("language")
+        video_id = self.match.group("video_id")
 
-        if not video["videoJsonPlayer"]["VSR"]:
+        json_url = self.API_URL.format(language, video_id or "LIVE")
+        headers = {
+            "Authorization": f"Bearer {self.API_TOKEN}"
+        }
+        streams, metadata = self.session.http.get(json_url, headers=headers, schema=validate.Schema(
+            validate.parse_json(),
+            {"data": {"attributes": {
+                "streams": validate.any(
+                    [],
+                    [
+                        validate.all(
+                            {
+                                "url": validate.url(),
+                                "slot": int,
+                                "protocol": validate.any("HLS", "HLS_NG"),
+                            },
+                            validate.union_get("slot", "protocol", "url")
+                        )
+                    ]
+                ),
+                "metadata": {
+                    "title": str,
+                    "subtitle": validate.any(None, str)
+                }
+            }}},
+            validate.get(("data", "attributes")),
+            validate.union_get("streams", "metadata")
+        ))
+
+        if not streams:
             return
 
-        vsr = video["videoJsonPlayer"]["VSR"]
-        return self._create_stream(vsr)
+        self.title = f"{metadata['title']} - {metadata['subtitle']}" if metadata["subtitle"] else metadata["title"]
+
+        for slot, protocol, url in sorted(streams, key=itemgetter(0)):
+            return HLSStream.parse_variant_playlist(self.session, url)
```

Comment:
Btw, I left the for-loop here because of the code removal for the sub-stream selection. It currently simply returns the first one sorted by the slot number.